compiler/rustc_mir_build/src/build/matches/test.rs

```diff
@@ -671,6 +671,16 @@ impl<'a, 'tcx> Builder<'a, 'tcx> {
             (&TestKind::Range { .. }, _) => None,
 
             (&TestKind::Eq { .. } | &TestKind::Len { .. }, _) => {
+                // We do call `test()` below to see what kind of test `match_pair` would require.
+                // If it is the same test as `test`, then we can just use `test`.
```

Comment:
These sentences are hard to read. Maybe it would be clearer without them: if not, I think they ought to be tweaked (there are too many "test"s).

Reply:
Better now?